react-toggle-switch-component/src/toggle-switch.jsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+class ToggleSwitch extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { button: false };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState({ button: !this.state.button });
+  }
+
+  render() {
+    if (this.state.button) {
+      return <button onClick={this.handleClick} className='on'><span className='switch switch-on'></span></button>;
+    } else {
+      return <button onClick={this.handleClick} className='off'><span className='switch switch-off'></span></button>;
+    }
```

Comment:
This would be hard to maintain and prone to bugs and inconsistencies. A better approach would be:
```JSX
let status = 'off';

if (this.state.button) {
  let status = 'on';
}

return (
  <button onClick={this.handleClick} className={status}>
    <span className={`switch switch-${status}`}></span>
  </button>
);
```